### DISCUSSION TITLE
Add site as a browser Search Engine

### DISCUSSION THREAD

**Discussion by SilentGlasses** [squidfunk/mkdocs-material#2713] [Ideas]
- [x] I've read the [contribution guidelines][1] and agree with them

__I want to suggest an idea and checked that ...__

- [x] ... to my best knowledge, my idea wouldn't break something for other users
- [x] ... the documentation does not mention anything about my idea
- [x] ... there are no open or closed issues that are related to my idea

## Description

Ability to use the site as a search engine.

### Use Cases

We have most of our internal sites like confluence and git added to browsers as optional search engines so that our employees can search sites without having to load the entire site. It would be great to be able to do this with our Mkdocs site.  

For example, for Google Drive we do something like this:

Search engine: Google Drive
Keyword: gdrive
URL: `https://drive.google.com/drive/search?q=%s`

So then a user would type `grdive` in the url press tab then enter their search string.

  [1]: https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md

**Comment by squidfunk** [comment]
Thank for suggesting! I already prototyped this as part of Insiders (the standard is called OpenSearch, right?) but couldn't get Chrome to do the <kbd>TAB</kbd> -> `Search <site name>`. In Firefox, I achieved parts of it by adding it explicitly as a search engine, but the UX was meh. Any further reading you can recommend?

**Comment by squidfunk** [comment]
Moving this to a discussion.
